guides/_posts/2015-01-06-constraints.html

```diff
@@ -0,0 +1,839 @@
+---
+uid: CCSS
+title: Constraint CSS
+subtitle: lay it out like it's 1999
+color: b
+layout: post
+---
+
+<ul class="large-list cols-2">
+  <li><a href="#basicconstraint">Constraints, the basics</li>
+  <li><a href="#selectors">Selectors</a></li>
+  <li><a href="#rulesets">Rulesets</a></li>
+  <li><a href="#scope">Scope</a></li>
+  <li><a href="#position">Positionning elements</a></li>
+  <li><a href="#virtuals">Virtuals</a></li>
+</ul>
+
+
+Soon after the W3C introduced Cascading Style Sheets, [Greg Badros](http://www.badros.com/greg/), the author of the Cassowary Constraint
+Solver & [recently retired Facebook VP](http://allthingsd.com/20130920/facebook-engineering-and-products-vp-greg-badros-to-leave-company/),
+proposed [Constraint CSS (CCSS)](http://www.cs.washington.edu/research/constraints/web/ccss-uwtr.pdf) as a general solution for CSS layout.
+Back in '99 Badros demonstrated *responsive* layouts with CCSS that today's designers still can't reproduce without grinding out piles of JavaScript.
+For more than a decade, no one seemed to take notice outside academia... Until Apple implemented Cassowary & Greg's pioneering concepts in its new
+AutoLayout engine with the launch of OS X Lion. Despite the evolutionary leap for app developers, web designer's have had to settle
+with [float-based](http://www.sitepoint.com/give-floats-the-flick-in-css-layouts/) & table-based layouts that have remained unimproved to this day.
+
+> "We should contemplate how very, very far behind the web platform is in making it delightful to build the sorts of things that are work-a-day in native environments." <a href="http://infrequently.org/2012/02/misdirection/">Alex Russell</a>
+
+The foundation of GSS is a modernized implementation of CCSS.  From CCSS primitives other, more exotic layout API's can and are accomplished.
+
+<a name="basicconstraint"></a>
+## Constraints, the basics
+
+Constraints express [relationships between variables](#constraintsaretwoways) that [may or may not hold](#mayofmaynothold). You can constrain any
+numerical style property of an element, not just its position & size.
+
+For example, if I want all paragraph tags to have line-height greater than 16px and less than 1/12th the size of the window:
+
+{% highlight css %}
+
+p[line-height] >= 16;
+p[line-height] <= ::window[height] / 12;
+
+{% endhighlight %}
+
+### Constraints declaration
+
+Let's go through the syntax for declaring a constraint:
+
+`p[line-height] >= 16;`
+
++ `p` is the selector. This constraint will apply to every `p` tag in the page.
++ `[]` is the style property selector.
++ `line-height` is the style property for which a value will be calculated by GSS.
++ `>=` defines an inequality constraint. You can declare inequality and equality constraint.
++ `16` is the numerical value for the constraint in pixel unit. The default unit in GSS are pixels.
```

Comment:
Something seems off here. Maybe it should be:

> `16` is the numerical value for the constraint in pixels (the default unit of measurement in GSS).

or something that doesn't mix singular and plural for "pixel", and "unit"